datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/AbstractKafkaBasedConnectorTask.java

```diff
@@ -436,11 +436,15 @@ public void run() {
         }
       }
       postShutdownHook();
-      _stoppedLatch.countDown();
+      releaseTaskLatch();
```

Comment:
releaseTaskLatch does not sound right. At Abstract level, there is no concept of releaseTask. How about markTaskAsDown or markTaskAsStopped ? Also, since you got rid of the other half, you can skip creation of function from this and do it together as second half.